core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -191,8 +191,8 @@ void *impl_allocate_common(const int device_id,
                 "CUDA runtime version >=12.2 required when "
                 "Kokkos_ENABLE_IMPL_CUDA_UNIFIED_MEMORY is set");
   if (arg_alloc_size) {  // cudaMemAdvise_v2 does not work with nullptr
-    ptr = malloc(arg_alloc_size);
-    if (ptr != nullptr) {
+    error_code = cudaMallocManaged(&ptr, arg_alloc_size, cudaMemAttachGlobal);
```

Comment:
Why not use `malloc`? 

`cudaMallocManaged` currently leads to very different behavior: UVM still implies page migration, while it is remotely accessed with `malloc` allocated memory.

Perhaps I missed something, but all the documents I have read explicitly stated that UVM and unified memory (ATS) are somewhat different.

@crtrott any comment?

Reply:
The code is marked as outdated, but it is still present; I've commented on the introducing commit.

A good read about GH memory: https://arxiv.org/html/2407.07850v1